pkg/types/v1/config.go

```diff
@@ -97,6 +97,18 @@ func (i *InstallSpec) Sanitize() error {
 	if extraPartsSizeCheck == 1 && i.Partitions.Persistent.Size == 0 {
 		return fmt.Errorf("both persistent partition and extra partitions have size set to 0. Only one partition can have its size set to 0 which means that it will take all the available disk space in the device")
 	}
+
+	// In case we got values override from the config files, we need to make sure to set the labels properly
+	// Otherwise we might break the system as we rely on know labels for a lot of things.
+	i.Partitions.OEM.FilesystemLabel = constants.OEMLabel
+	i.Partitions.OEM.Name = constants.OEMPartName
+	i.Partitions.State.FilesystemLabel = constants.StateLabel
+	i.Partitions.State.Name = constants.StatePartName
+	i.Partitions.Persistent.FilesystemLabel = constants.PersistentLabel
+	i.Partitions.Persistent.Name = constants.PersistentPartName
+	i.Partitions.Recovery.FilesystemLabel = constants.RecoveryLabel
+	i.Partitions.Recovery.Name = constants.RecoveryPartName
```

Comment:
Can we make this a method of partitions? would be nice to call it something like:

```
i.Partitions.SetDefaultLabels()
```

Feel free to choose a better name, but I think it reduces the size of the outer scope function and keeps the details on how to set the default inside partitions

Reply:
Good point. It started with one line and then it kept growing with more :D